components/style/values/specified/mod.rs

```diff
@@ -621,19 +621,15 @@ impl Shadow {
             if !lengths_parsed {
                 if let Ok(value) = input.try(|i| Length::parse(context, i)) {
                     lengths[0] = value;
-                    let mut length_parsed_count = 1;
-                    while length_parsed_count < length_count {
+
+                    lengths[1] = try!(Length::parse(context, input));
+
+                    if let Ok(value) = input.try(|i| Length::parse_non_negative(i)) {
+                        lengths[2] = value;
+
                         if let Ok(value) = input.try(|i| Length::parse(context, i)) {
-                            lengths[length_parsed_count] = value
-                        } else {
-                            break
+                            lengths[3] = value;
```

Comment:
That makes it not return non-zero spread in that case, but doesn't block the function from parsing invalid value.